src/shell/execute.rs

```diff
@@ -589,12 +589,15 @@ async fn execute_subshell(
   )
   .await;
 
-  // sub shells do not cause an exit
   match result {
     ExecuteResult::Exit(code, handles) => {
+      // sub shells do not cause an exit
+      ExecuteResult::Continue(code, Vec::new(), handles)
+    }
+    ExecuteResult::Continue(code, _env_changes, handles) => {
+      // env changes are not propagated
       ExecuteResult::Continue(code, Vec::new(), handles)
     }
-    ExecuteResult::Continue(_, _, _) => result,
```

Comment:
I just went to go fix this in dax and interestingly this was impelmented correctly there, but not here.